src/PortAudio.jl

```diff
@@ -730,29 +730,35 @@ function PortAudioStream(
                     output_device.output_bounds.high_latency,
                 )
             end
-            if samplerate === nothing
-                samplerate = combine_default_sample_rates(
+            samplerate = if samplerate === nothing
+                combine_default_sample_rates(
                     input_device,
                     input_device.default_sample_rate,
                     output_device,
                     output_device.default_sample_rate,
                 )
+            else
+                float(samplerate)
             end
         else
             if latency === nothing
                 latency = input_device.input_bounds.high_latency
             end
-            if samplerate === nothing
-                samplerate = input_device.default_sample_rate
+            samplerate = if samplerate === nothing
+                input_device.default_sample_rate
+            else
+                float(samplerate)
             end
         end
     else
         if output_channels_filled > 0
             if latency === nothing
                 latency = output_device.output_bounds.high_latency
             end
-            if samplerate === nothing
-                samplerate = output_device.default_sample_rate
+            samplerate = if samplerate === nothing
+                output_device.default_sample_rate
+            else
+                float(samplerate)
             end
```

Comment:
```suggestion
                    output_device.default_sample_rate
                else
                    float(samplerate)
                end
```
It seems unusual to me to have `else` and `end` in a column without a matching `if` in the same column.
But it's not that important so if you prefer your original way it's ok.

Reply:
Doesn't matter much to be but ok.